### PR TITLE
Chart context menu and Copy to Clipboard

### LINKED ISSUE #2389: Charts should provide extended functionality via context menu
We may have need for this in a client app, previously implemented in Ext.

The context menu provides the ability to:
 
1. Copy the chart to the clipboard in varying sizes.
2. Export the chart as a png file
3. Export the series data as an Excel spreadsheet
4. Print the chart

### PR DESCRIPTION
**Pull request by cnrudd**
This PR resolves #2389, and brings in code that adds a context menu to charts (enabled by default) with some default menu items.
It also adds a new Highcharts extension "copyToClipboard", which can be used by the context menu or export menu button.

This "copyToClipboard" is an iteration on our previous implementation in that it no longer requires a server round trip, and requires only 1 click, compared with 2 before.  It only works on webkit based browsers.   On non-webkit browsers, the "Copy to Clipboard" menu item does not appear.

The following are examples of showContextMenu declarations on a chartModel with the results expected.
```
        // showContextMenu: undefined                   // => show default   // confirmed
        // showContextMenu: null                        // => show none      // confirmed
        // showContextMenu: false                       // => show none      // confirmed
        // showContextMenu: true                        // => show default   // confirmed
```

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile: not required
- [x] Created Toolbox branch / PR:  not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: cmp/chart/impl/copyToClipboard.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+/**
+ * Copy the chart in it's current state to the clipboard.
+ * Works only on webkit based browsers.
+ */
+export function installCopyToClipboard(Highcharts) {
+    if (!Highcharts) return;
+    const  {Chart, extend} = Highcharts;
+
+    extend(Chart.prototype, {
+        copyToClipboard: async function(exportingOptions, chartOptions) {
```
Comment (lbwexler) [inline cmp/chart/impl/copyToClipboard.js:19]: put implementation function also outside of the extend, for clarity.

File: cmp/chart/impl/copyToClipboard.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+ * @param {Highcharts.Chart} chart
+ * @param {Highcharts.Exporting.ExportingOptions} exportingOptions
+ * @param {Highcharts.Options} chartOptions
+ * @return {blob}
+ */
+async function convertChartToPngAsync(chart, exportingOptions, chartOptions) {
+    const svg = await new Promise(
+            (resolve, reject)  => chart.getSVGForLocalExport(
+                merge(chart.options.exporting, exportingOptions),
+                chartOptions || {},
```
Comment (lbwexler) [inline cmp/chart/impl/copyToClipboard.js:56]: use `??`

Comment (cnrudd) [reply cmp/chart/impl/copyToClipboard.js]: done.

File: cmp/chart/impl/copyToClipboard.js (hunk 1 above)
Comment (cnrudd) [reply cmp/chart/impl/copyToClipboard.js]: done.

**Pull request merged**